From ca0d4de08866a068a13203cae65a3f8fe4c837ed Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Thu, 10 Jul 2014 01:07:09 -0400
Subject: [acpi] hotplug/pci: Fix error code path in acpiphp_enumerate_slots()

Message-id: <20140710010709.22146.1672.stgit@gir.stowe>
Patchwork-id: 85625
O-Subject: [RHEL7.1 PATCH 086/105] ACPI / hotplug / PCI: Fix error code path in acpiphp_enumerate_slots()
Bugzilla: 1114228
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1114228
Upstream Status: 5d4494573c1fed806b97c2a1ba619d50bb89d1ed

commit 5d4494573c1fed806b97c2a1ba619d50bb89d1ed
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Fri Oct 11 13:20:50 2013 +0200

    ACPI / hotplug / PCI: Fix error code path in acpiphp_enumerate_slots()

    One of the error code paths in acpiphp_enumerate_slots() is missing
    a pci_dev_put(bridge->pci_dev) call, so add it.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Acked-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index 6557702..7b5166d 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -1003,6 +1003,7 @@ void acpiphp_enumerate_slots(struct pci_bus *bus)
   if (WARN_ON(!context)) {
    mutex_unlock(&acpiphp_context_lock);
    put_device(&bus->dev);
+   pci_dev_put(bridge->pci_dev);
    kfree(bridge);
    return;
   }
-- 
1.7.1